tests/scripts/check_names.py

```diff
@@ -249,27 +249,39 @@ def comprehensive_parse(self):
             .format(str(self.excluded_files))
         )
 
-        all_macros = self.parse_macros([
+        public_macros = self.parse_macros([
```

Comment:
Partly preexisting: we should make a list of public and private patterns, rather than repeat them with unexplained discrepancies (like the inclusion of `psa` and `test/drivers`.

Reply:
Can you please elaborate what change you are suggesting, and if it is a blocker for this PR?

Reply:
```
PUBLIC_HEADERS = […]
PRIVATE_HEADERS = […]
```
then
```
public_macros = self.parse_macros(PUBLIC_HEADERS)
```
and so on.

Please fix this — it's preexisting but your patch makes the duplication worse.